src/base/web3-connection.ts

```diff
@@ -66,7 +71,9 @@ export class Web3Connection {
     const web3Link = web3Host.toLowerCase();
     let provider: HttpProvider|IpcProvider|WebsocketProvider;
 
-    if (web3Link.includes(`http`))
+    if (this.options.provider)
```

Comment:
the start function can't not have been started before _and_ have an options.provider -- if this happens we're making some weird usage of this class and we should instead return on the line 58

Reply:
removed as agreed on discord